modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/java/com/liferay/dynamic/data/mapping/web/internal/item/selector/DDMFormInstanceItemSelectorView.java

```diff
@@ -0,0 +1,345 @@
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+import javax.portlet.PortletRequest;
+import javax.portlet.PortletURL;
+
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Mikel Lorza
+ */
+@Component(
+	property = "item.selector.view.order:Integer=200",
+	service = ItemSelectorView.class
+)
+public class DDMFormInstanceItemSelectorView
+	implements InfoItemSelectorView,
+			   ItemSelectorView<InfoItemItemSelectorCriterion> {
+
+	@Override
+	public String getClassName() {
+		return DDMFormInstance.class.getName();
+	}
+
+	@Override
+	public Class<InfoItemItemSelectorCriterion>
+		getItemSelectorCriterionClass() {
+
+		return InfoItemItemSelectorCriterion.class;
+	}
+
+	@Override
+	public List<ItemSelectorReturnType> getSupportedItemSelectorReturnTypes() {
+		return _supportedItemSelectorReturnTypes;
+	}
+
+	@Override
+	public String getTitle(Locale locale) {
+		return _language.get(locale, "forms");
+	}
+
+	@Override
+	public void renderHTML(
+			ServletRequest servletRequest, ServletResponse servletResponse,
+			InfoItemItemSelectorCriterion infoItemItemSelectorCriterion,
+			PortletURL portletURL, String itemSelectedEventName, boolean search)
+		throws IOException, ServletException {
+
+		_itemSelectorViewDescriptorRenderer.renderHTML(
+			servletRequest, servletResponse, infoItemItemSelectorCriterion,
+			portletURL, itemSelectedEventName, search,
+			new DDMFormInstanceSelectorViewDescriptor(
+				(HttpServletRequest)servletRequest,
+				infoItemItemSelectorCriterion, portletURL));
+	}
+
+	private static final List<ItemSelectorReturnType>
+		_supportedItemSelectorReturnTypes = Collections.singletonList(
+			new InfoItemItemSelectorReturnType());
+
+	@Reference
+	private DDMFormInstanceService _ddmFormInstanceService;
+
+	@Reference
+	private ItemSelectorViewDescriptorRenderer<InfoItemItemSelectorCriterion>
+		_itemSelectorViewDescriptorRenderer;
+
+	@Reference
+	private Language _language;
+
+	@Reference
+	private Portal _portal;
+
+	private class DDMFormInstanceItemDescriptor
+		implements ItemSelectorViewDescriptor.ItemDescriptor {
+
+		public DDMFormInstanceItemDescriptor(
+			DDMFormInstance ddmFormInstance,
+			HttpServletRequest httpServletRequest) {
+
+			_ddmFormInstance = ddmFormInstance;
+			_httpServletRequest = httpServletRequest;
+
+			_resourceBundle = ResourceBundleUtil.getBundle(
+				httpServletRequest.getLocale(), getClass());
+		}
+
+		@Override
+		public String getIcon() {
+			return "forms";
+		}
+
+		@Override
+		public String getImageURL() {
+			return null;
+		}
+
+		@Override
+		public Date getModifiedDate() {
+			return _ddmFormInstance.getModifiedDate();
+		}
+
+		@Override
+		public String getPayload() {
```

Comment:
No, as I can figure out , we must to implement the payload because we are using the default item selector renderer impl and with this renderer we have to create an item selector view descriptor and inside it we have to implement an item selector to manage the data of the each item. So, yes, we have to implement our payload.